src/platform/backends/qemu/linux/qemu_platform_detail_linux.cpp

```diff
@@ -0,0 +1,180 @@
+    {
+        mpl::log(mpl::Level::warning, category,
+                 fmt::format("Failed to write to {}", qUtf8Printable(ip_forward.fileName())));
+    }
+}
+
+mp::DNSMasqServer::UPtr init_nat_network(const mp::Path& network_dir, const QString& bridge_name,
+                                         const std::string& subnet)
+{
+    create_virtual_switch(subnet, bridge_name);
+    set_ip_forward();
+
+    return MP_DNSMASQ_SERVER_FACTORY.make_dnsmasq_server(network_dir, bridge_name, subnet);
+}
+
+void delete_virtual_switch(const QString& bridge_name)
+{
+    if (MP_UTILS.run_cmd_for_status("ip", {"addr", "show", bridge_name}))
+    {
+        MP_UTILS.run_cmd_for_status("ip", {"link", "delete", bridge_name});
+    }
+}
+} // namespace
+
+mp::QemuPlatformDetail::QemuPlatformDetail(const mp::Path& data_dir)
+    : bridge_name{multipass_bridge_name},
+      network_dir{mp::utils::make_dir(QDir(data_dir), "network")},
+      subnet{MP_BACKEND.get_subnet(network_dir, bridge_name)},
+      dnsmasq_server{init_nat_network(network_dir, bridge_name, subnet)},
+      firewall_config{MP_FIREWALL_CONFIG_FACTORY.make_firewall_config(bridge_name, subnet)}
+{
+}
+
+mp::QemuPlatformDetail::~QemuPlatformDetail()
+{
+    delete_virtual_switch(bridge_name);
+}
+
+mp::optional<mp::IPAddress> mp::QemuPlatformDetail::get_ip_for(const std::string& hw_addr)
+{
+    return dnsmasq_server->get_ip_for(hw_addr);
+}
+
+void mp::QemuPlatformDetail::remove_resources_for(const std::string& name)
+{
+    auto it = name_to_net_device_map.find(name);
+    if (it != name_to_net_device_map.end())
+    {
+        const auto& [tap_device_name, hw_addr] = it->second;
+        dnsmasq_server->release_mac(hw_addr);
+        remove_tap_device(tap_device_name);
+
+        name_to_net_device_map.erase(name);
+    }
+}
+
+void mp::QemuPlatformDetail::platform_health_check()
+{
+    MP_BACKEND.check_for_kvm_support();
+    MP_BACKEND.check_if_kvm_is_in_use();
+
+    dnsmasq_server->check_dnsmasq_running();
+    firewall_config->verify_firewall_rules();
+}
+
+QStringList mp::QemuPlatformDetail::qemu_platform_args(const VirtualMachineDescription& vm_desc)
+{
+    // Configure and generate the args for the default network interface
+    auto tap_device_name = generate_tap_device_name(vm_desc.vm_name);
+    create_tap_device(tap_device_name, bridge_name);
+
+    name_to_net_device_map.emplace(vm_desc.vm_name, std::make_pair(tap_device_name, vm_desc.default_mac_address));
+
+    return QStringList() << "--enable-kvm"
+                         // Pass host CPU flags to VM
+                         << "-cpu"
+                         << "host"
+                         // Set up the network related args
+                         << "-nic"
+                         << QString::fromStdString(
+                                fmt::format("tap,ifname={},script=no,downscript=no,model=virtio-net-pci,mac={}",
+                                            tap_device_name, vm_desc.default_mac_address));
```

Comment:
So `-nic` is `-device ... -netdev ...` in one go?

Reply:
Yes, I just discovered that and it works nicely.  https://www.qemu.org/2018/05/31/nic-parameter/ for reference.